dev/tests/api-functional/testsuite/Magento/Quote/Api/CartAddingItemsTest.php

```diff
@@ -92,4 +99,86 @@ public function testPriceForCreatingQuoteFromEmptyCart()
         $quote->load($quoteId);
         $quote->delete();
     }
+
+    /**
+     * Test qty for cart after adding grouped product with custom qty.
+     *
+     * @magentoApiDataFixture Magento/GroupedProduct/_files/product_grouped_with_simple.php
+     * @magentoApiDataFixture Magento/Customer/_files/customer_one_address.php
+     * @return void
+     */
+    public function testAddToCartGroupedCustomQuantity(): void
+    {
+        $this->_markTestAsRestOnly();
+
+        $firstProductId = $this->productResource->getIdBySku('simple_11');
+        $secondProductId = $this->productResource->getIdBySku('simple_22');
+        $qtyData = [$firstProductId => 2, $secondProductId => 4];
+
+        // Get customer ID token
+        /** @var CustomerTokenServiceInterface $customerTokenService */
+        $customerTokenService = $this->objectManager->create(CustomerTokenServiceInterface::class);
+        $token = $customerTokenService->createCustomerAccessToken(
+            'customer_one_address@test.com',
+            'password'
+        );
+
+        // Creating empty cart for registered customer.
+        $serviceInfo = [
+            'rest' => [
+                'resourcePath' => '/V1/carts/mine',
+                'httpMethod' => Request::HTTP_METHOD_POST,
+                'token' => $token
+            ]
+        ];
+
+        $quoteId = $this->_webApiCall($serviceInfo, ['customerId' => 999]); // customerId 999 will get overridden
+        $this->assertGreaterThan(0, $quoteId);
+
+        // Adding item to the cart
+        $productOptionData = [
+            'extension_attributes' => [
+                'grouped_options' => [
+                    ['id' => $firstProductId, 'qty' => $qtyData[$firstProductId]],
+                    ['id' => $secondProductId, 'qty' => $qtyData[$secondProductId]],
```

Comment:
Should these be using `sku` instead of `id` (and/or allow the ability to use either `sku` or `id`) to match the other aspects of the API? Even the `cartItem` above passes in a `sku`. This would seem like the call might look like:
```
{
  "cartItem": {
    "sku": "my_group_1",
    "qty": 1,
    "quote_id": "4558",
    "product_option": {
      "extension_attributes": {
        "grouped_options": [
          {
            "sku": "sku_1",
            "qty": 1
          },
          {
            "sku": "sku_2",
            "qty": 1
          }
        ]
      }
    }
  }
}
```